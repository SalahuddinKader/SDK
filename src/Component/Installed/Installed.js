import React from "react";
import { Box, Card, CardActions, CardContent, Typography } from "@mui/material";
import { orderBy, groupBy } from "lodash";
import moment from "moment";
import InstalledData from "../../installed.json";
const { installedSdks } = InstalledData.data;
/**
 * installed js
 * @func {function} renderData // groupBy by lodash and map through the list
 */
const Installed = () => {
  const newInstalledSdks = orderBy(installedSdks, "categories", "asc"); //instaslledSdk orderBy lodash

  //Handling categories by group and map through the installedSdklist
  const renderData = () => {
    const data = groupBy(newInstalledSdks, "categories");
    return Object.keys(data).map((category) => (
      <React.Fragment key={category}>
        <Box
          sx={{
            marginTop: "20px",
            color: "#42a5f5",
          }}
        >
          <Typography variant="h6" sx={{ textDecoration: "underline" }}>
            {category}
          </Typography>

          {data[category].map((order, i) => {
            const { lastSeenDate, name } = order;
            //Used moment to get currentDate and lastSeenDate
            const currentDate = moment(new Date());
            const lastSeenInstalledDate = moment(lastSeenDate);
            return (
              <Typography key={i}>
                {name}{" "}
                {`${Math.floor(
                  currentDate.diff(lastSeenInstalledDate, "years", true)
                )} years ago`}
              </Typography>
            );
          })}
        </Box>
      </React.Fragment>
    ));
  };

  return (
    <Card
      sx={{
        width: "50%",
        border: "3px solid #42a5f5",
        height: "100%",
        overflow: "hidden",
        margin: "10px",
        padding: "10px",
      }}
    >
      <CardContent>
        <Typography color="#42a5f5">Installed SDK's</Typography>

        {/* Display length of SDK's */}
        <Typography
          sx={{
            display: "flex",
            justifyContent: "flex-end",
            flexDirection: "row",
          }}
          color="#42a5f5"
        >
          {installedSdks.length}
        </Typography>

        {/* Latest Update date */}
        <Typography color="#42a5f5">
          {` Latest Update: ${moment
            .utc(installedSdks[0].lastSeenDate)
            .local()
            .format("ll")}`}
        </Typography>
        <Typography
          sx={{ width: "100%", mt: 1.5, borderBottom: "1px solid #42a5f5" }}
          color="#42a5f5"
        ></Typography>

        {/* Call renderData to display installedSdk's list */}
        <Box>{renderData()}</Box>
      </CardContent>
      <CardActions></CardActions>
    </Card>
  );
};

export default Installed;
