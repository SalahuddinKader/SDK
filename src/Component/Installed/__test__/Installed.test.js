import React from "react";
import ReactDom from "react-dom";
import Installed from "../Installed";

import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";

it("renders without crashing", () => {
  const Box = document.createElement("Box");
  ReactDom.render(<Installed></Installed>, Box);
});

it("matches snapshort", () => {
  const tree = renderer.create(<Installed></Installed>).toJSON();
  expect(tree).toMatchSnapshot();
});
