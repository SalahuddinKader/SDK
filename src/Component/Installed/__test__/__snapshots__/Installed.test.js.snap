// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshort 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1ocap4b-MuiPaper-root-MuiCard-root"
>
  <div
    className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 css-cmfhp1-MuiTypography-root"
    >
      Installed SDK's
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 css-jrcnce-MuiTypography-root"
    >
      10
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 css-cmfhp1-MuiTypography-root"
    >
       Latest Update: Jul 16, 2019
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 css-1rnp0n2-MuiTypography-root"
    />
    <div
      className="MuiBox-root css-0"
    >
      <div
        className="MuiBox-root css-17s8iu9"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6 css-133mrz3-MuiTypography-root"
        >
          App Platform
        </h6>
        <p
          className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          SDK 4
           
          2 years ago
        </p>
      </div>
      <div
        className="MuiBox-root css-17s8iu9"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6 css-133mrz3-MuiTypography-root"
        >
          Backend
        </h6>
        <p
          className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          SDK 1
           
          2 years ago
        </p>
      </div>
      <div
        className="MuiBox-root css-17s8iu9"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6 css-133mrz3-MuiTypography-root"
        >
          Location
        </h6>
        <p
          className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          SDK 7
           
          2 years ago
        </p>
      </div>
      <div
        className="MuiBox-root css-17s8iu9"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6 css-133mrz3-MuiTypography-root"
        >
          OCR
        </h6>
        <p
          className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          SDK 9
           
          2 years ago
        </p>
      </div>
      <div
        className="MuiBox-root css-17s8iu9"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6 css-133mrz3-MuiTypography-root"
        >
          Payments
        </h6>
        <p
          className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          SDK 3
           
          2 years ago
        </p>
      </div>
      <div
        className="MuiBox-root css-17s8iu9"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6 css-133mrz3-MuiTypography-root"
        >
          Social
        </h6>
        <p
          className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          SDK 6
           
          2 years ago
        </p>
      </div>
      <div
        className="MuiBox-root css-17s8iu9"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6 css-133mrz3-MuiTypography-root"
        >
          Utilities
        </h6>
        <p
          className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          SDK 2
           
          2 years ago
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          SDK 5
           
          2 years ago
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          SDK 8
           
          2 years ago
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          SDK 10
           
          2 years ago
        </p>
      </div>
    </div>
  </div>
  <div
    className="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
  />
</div>
`;
